examples/so_vits_svc.py

```diff
@@ -0,0 +1,834 @@
+    qk = q @ k
+    if mask is not None: qk = qk + mask[:n_ctx, :n_ctx]
+    w = qk.softmax(-1)
+    return (w @ v).permute(0, 2, 1, 3).flatten(start_dim=2), qk.detach()
+
+class ConvFeatureExtractionModel():
+  def __init__(self, conv_layers, dropout=.0, mode="default", conv_bias=False):
+    assert mode in {"default", "group_norm_masked", "layer_norm"}
+    self.mode = mode
+    def block(n_in, n_out, k, stride, is_layer_norm=False, is_group_norm=False, conv_bias=False):
+      def make_conv():
+        conv = nn.Conv1d(n_in, n_out, k, stride=stride, bias=conv_bias)
+        conv.weight = Tensor.kaiming_normal(*conv.weight.shape)
+        return conv
+      class SequentialMasked(Sequential):  # https://github.com/auspicious3000/contentvec/blob/main/contentvec/models/wav2vec/wav2vec2_1.py#L56
+        def __init__(self, list): super().__init__(list)
+        def __call__(self, x, mask):
+          x = self.list[0](x)
+          x = self.list[1](x)
+          x = self.list[2](x, mask)
+          x = self.list[3](x)
+          return x
+      assert (is_layer_norm and is_group_norm) == False, "layer norm and group norm are exclusive"
+      if is_layer_norm:
+        return Sequential([make_conv(), Dropout(p=dropout), Sequential([TransposeLast(), Fp32LayerNorm(dim, elementwise_affine=True), TransposeLast()]), GELU()])
+      elif is_group_norm and mode == "default":
+        return Sequential([make_conv(), Dropout(p=dropout), Fp32GroupNorm(dim, dim, affine=True), GELU()])
+      elif is_group_norm and mode == "group_norm_masked":
+        return SequentialMasked([make_conv(), Dropout(p=dropout), GroupNormMasked(dim, dim, affine=True), GELU()])
+      else:
+        return Sequential([make_conv(), Dropout(p=dropout), GELU()])
+    in_d = 1
+    self.conv_layers = []
+    for i, cl in enumerate(conv_layers):
+      assert len(cl) == 3, "invalid conv definition: " + str(cl)
+      (dim, k, stride) = cl
+      if i == 0:
+        self.cl = cl
+      self.conv_layers.append(block(in_d, dim, k, stride, is_layer_norm=(mode == "layer_norm"), is_group_norm=((mode == "default" or mode == "group_norm_masked") and i == 0), conv_bias=conv_bias))
+      in_d = dim
+
+  def __call__(self, x:Tensor, padding_mask:Tensor):  # TODO: refactor this later
+    x = x.unsqueeze(1)  # BxT -> BxCxT
+    for i, conv in enumerate(self.conv_layers) :
+      if i == 0:
+        if self.mode == "group_norm_masked":
+          if padding_mask is not None:
+            _, k, stride = self.cl
+            lengths_org = tilde(padding_mask.cast(dtypes.bool)).cast(dtypes.int64).sum(1)  # ensure padding_mask is bool for tilde
+            lengths = (((lengths_org - k) / stride) + 1).floor().cast(dtypes.int64)
+            padding_mask = tilde(lengths_to_padding_mask(lengths)).cast(dtypes.int64)  # lengths_to_padding_mask returns bool tensor
+          x = conv(x, padding_mask)  # padding_mask is numeric
+        else:
+          x = conv(x)  # default
```

Comment:
can we just move the first iteration out of the loop? enumerate has a start parameter.

Reply:
[9109e9e](https://github.com/tinygrad/tinygrad/pull/1371/commits/9109e9ee3b8e7a875963a6abd36bc84ae9090fbf)